src/AudioAnalysisTools/Harmonics/HarmonicParameters.cs

```diff
@@ -189,23 +207,17 @@ public class HarmonicParameters : CommonParameters
             var intensity = new double[rowCount];
             var maxIndexArray = new int[rowCount];
 
-            // for all time frames
-            for (int t = 2; t < rowCount - 2; t++)
+            // Run a moving average filter along each frequency bin.
+            // This may help to fill noise gaps in formants. Ignore values <3.
+            if (smoothingWindow > 2)
```

Comment:
```suggestion
            if (smoothingWindow is not null)
```

assuming you take the above suggestion

Reply:
It is meant to be >2 because there is no point doing a moving average with window = 0 or 1. And the best minimum value if it is to be done is 3.

Reply:
Right, there are two concerns, _should we smooth?_ and _is the smoothing value valid?_

If you add the validation from the comment above, we can ensure a bad value (`<3`) is not supplied by the user.

```suggestion
            if (smoothingWindow is not null) {
               var smoothing = Math.Max(3, smoothingWindow.Value);
```

```suggestion
            if (smoothingWindow is not null) {
                if (smoothingWindow.Value < 3) { throw new ArgumentException("cannot be less than 3", nameof(smoothingWindow)' }
```